### PR TITLE
Added BDD User login sentences/implementation

### PR DESCRIPTION
**Pull request by miguelcleverti**
Added implementation for user with diferent roles login.
relates to jira issue: https://jira.ez.no/browse/EZP-24753

depends on: ezsystems/BehatBundle#33

### CODE REVIEW

**Comment by miguelcleverti** [comment]
@andrerom @joaoinacio @pcardiga

File: Features/Context/SubContext/Authentication.php [hunk 1]
```diff
@@ -59,6 +58,33 @@ public function loggedAsUserPlatformUi()
     }
 
     /**
+     * @Given I am logged in as an :role in PlatformUI
+     */
+    public function loggedAsEditorPlatformUI($role)
+    {
+        switch ($role) {
```
Comment (joaoinacio) [inline Features/Context/SubContext/Authentication.php]: I would probably add it in a separate function
````php
/**
 * Returns default user group name from role name. 
 */
public function getUserGroupForRolle($role)
```

**Comment by joaoinacio** [comment]
Regardless of my inline comment, EzContext should already provide the means to create and return a user/password for a specific role:
```php
    /**
     * Get credentials for a specific role
     *
     * @uses \EzSystems\BehatBundle\Context\Object\User
     * @throws \eZ\Publish\API\Repository\Exceptions\NotFoundException
     *
     * @param string $role Role intended for testing
     *
     * @return array Associative with 'login' and 'password'
     */
    protected function getCredentialsFor( $roleIdentifier )
```
See https://github.com/ezsystems/BehatBundle/blob/master/Context/EzContext.php#L243

**Comment by miguelcleverti** [comment]
@andrerom made the refactor suggested by @joaoinacio

File: Features/Context/SubContext/Authentication.php [hunk 2]
```diff
@@ -51,7 +51,6 @@ public function goToPlatformUiAndLogIn($username, $password)
 
     /**
      * @Given I am logged in as admin on PlatformUI
-     * @Given I am logged in as an Editor in PlatformUI
```
Comment (joaoinacio) [inline Features/Context/SubContext/Authentication.php]: +1 after checking that this sentence is no longer being used :)

Comment (miguelcleverti) [inline Features/Context/SubContext/Authentication.php]: @joaoinacio checked

File: Features/Authentication/UsersLogin.feature [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+
+    @javascript
+    Scenario: As a Partner User, I want login to PlatformUI
+        Given I am logged in as an 'Partner' in PlatformUI
+        Then I should see "Welcome to eZ Platform" title
+
+    @javascript
+    Scenario: As a Subscriber User, I want login to PlatformUI
+        Given I am logged in as an 'Subscriber' in PlatformUI
+        Then I should see "Welcome to eZ Platform" title
```
Comment (andrerom) [inline Features/Authentication/UsersLogin.feature]: Where id you get this from? neither Subscriber, Member, Partner or Anonymous should have access to login, so here the spec is wrong, and maybe also impl, but that is a different issue.

Comment (miguelcleverti) [inline Features/Authentication/UsersLogin.feature]: @andrerom I made these tests myself, to test the Issue because that is what the fix does it makes possible all the roles login.

Comment (andrerom) [inline Features/Authentication/UsersLogin.feature]: Yes, but then use some tag for specs that are wrong, cause this is a bug, and you should not reflect bugs in specs.

Comment (andrerom) [inline Features/Authentication/UsersLogin.feature]: Suggested convention, when a spec specifies a bug
tag it with `@bug` and add link to JIRA issue for it, in this case it's all one bug, so the sections here can point to the same bug.

Comment (miguelcleverti) [inline Features/Authentication/UsersLogin.feature]: @andrerom so we should add the `@bug` tag instead of removing these scenarios? I think we should remove them or add the inverse test "U should not be able to log in" even if they will fail for now.

Comment (andrerom) [inline Features/Authentication/UsersLogin.feature]: That would be my preference, but then you should split it into separate PR linked to from the issue reported for this bug ;)

**Comment by miguelcleverti** [comment]
@andrerom removed the users that were not suppose to login

**Comment by andrerom** [comment]
+1 ready for merge? Is it passing? :)

**Comment by miguelcleverti** [comment]
@andrerom yes they are passing :smile:
